homeassistant/components/sharkiq/__init__.py

```diff
@@ -0,0 +1,131 @@
+"""Shark IQ Integration."""
+
+import asyncio
+
+import async_timeout
+from sharkiqpy import (
+    AylaApi,
+    SharkIqAuthError,
+    SharkIqAuthExpiringError,
+    SharkIqNotAuthedError,
+    get_ayla_api,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import API_TIMEOUT, COMPONENTS, DOMAIN, LOGGER
+from .update_coordinator import SharkIqUpdateCoordinator
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+
+class CannotConnect(exceptions.HomeAssistantError):
+    """Error to indicate we cannot connect."""
+
+
+async def async_setup(hass, config):
+    """Set up the sharkiq environment."""
+    hass.data.setdefault(DOMAIN, {})
+    if DOMAIN not in config:
+        return True
+    for index, conf in enumerate(config[DOMAIN]):
+        LOGGER.debug(
+            "Importing Shark IQ #%d (Username: %s)", index, conf[CONF_USERNAME]
+        )
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data=conf,
+            )
+        )
+
+
+async def async_connect_or_timeout(ayla_api: AylaApi) -> AylaApi:
+    """Connect to vacuum."""
+    try:
+        with async_timeout.timeout(API_TIMEOUT):
+            LOGGER.debug("Initialize connection to Ayla networks API")
+            await ayla_api.async_sign_in()
+    except SharkIqAuthError as exc:
+        LOGGER.error("Error to connect to Shark IQ api")
+        raise CannotConnect from exc
```

Comment:
This is an authentication error and not a cannot connect error. You should include that in the logged error message and return `False` so that `async_setup_entry` stops.

Reply:
To clarify, do you mean that I should return false instead of raising an exception?